reconcile/statuspage/atlassian.py

```diff
@@ -352,21 +338,21 @@ def _bind_component(
         if not dry_run:
             self._binding_state.bind_component(component_name, component_id)
 
-
-def init_provider_for_page(
-    page: StatusPageV1,
-    token: str,
-    component_binding_state: ComponentBindingState,
-) -> AtlassianStatusPageProvider:
-    """
-    Initializes the provider for atlassian status page.
-    """
-    return AtlassianStatusPageProvider(
-        page_name=page.name,
-        api=AtlassianRESTAPI(
-            page_id=page.page_id,
-            api_url=page.api_url,
-            token=token,
-        ),
-        component_binding_state=component_binding_state,
-    )
+    @staticmethod
+    def init_from_page(
+        page: StatusPageV1,
+        token: str,
+        component_binding_state: ComponentBindingState,
+    ) -> "AtlassianStatusPageProvider":
+        """
+        Initializes the provider for atlassian status page.
+        """
+        return AtlassianStatusPageProvider(
```

Comment:
This can be a classmethod and return `cls(...)`, return type annotation can use `Self` instead of class string.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4343/commits/7549eabf004bae065958b41195a124820ee1ea8e